src/Nervos/CommonFunc.h

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "CellDep.h"
+#include "OutPoint.h"
+#include "../BinaryCoding.h"
+#include "../Data.h"
+#include "../HexCoding.h"
+
+namespace TW::Nervos {
+
+enum AddressType {
+    FullVersion = 0,  // full version identifies the hash_type
+    HashIdx = 1,      // short version for locks with popular codehash, deprecated
+    DataCodeHash = 2, // full version with hash type 'Data', deprecated
+    TypeCodeHash = 4, // full version with hash type 'Type', deprecated
+};
+
+enum HashType { Data0 = 0, Type1 = 1, Data1 = 2 };
+
+class CommonFunc {
```

Comment:
can you move this to a new `Serialization.h` or `BinaryCoding.h`?

Reply:
Done